Content.Server/Traits/Assorted/EmotionLoopComponent.cs

```diff
@@ -0,0 +1,34 @@
+namespace Content.Server.Traits.Assorted;
+
+using System.Numerics;
+using Content.Shared.Chat.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+
+/// <summary>
+/// This component allows triggering any emotion at random intervals.
+/// </summary>
+[RegisterComponent, Access(typeof(EmotionLoopSystem))]
+public sealed partial class EmotionLoopComponent : Component
+{
+    /// <summary>
+    /// A minimum interval between emotions.
+    /// </summary>
+    [DataField("minTimeBetweenEmotions", required: true)]
+    public TimeSpan MinTimeBetweenEmotions { get; private set; }
+
+    /// <summary>
+    /// A maximum interval between emotions.
+    /// </summary>
+    [DataField("maxTimeBetweenEmotions", required: true)]
+    public TimeSpan MaxTimeBetweenEmotions { get; private set; }
+
+    /// <summary> 
+    /// A set of emotes that will be randomly picked from. 
+    /// <see cref="EmotePrototype"/> 
+    /// </summary> 
+    [DataField("emotes", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<EmotePrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<string> Emotes = new();
```

Comment:
Sorry about the last suggestion, as this should have the same effect while also being less confusing. I doubt the difference between using a HashSet verses a List will be that large, so this would be fine for our purposes.
 
```suggestion
    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public List<ProtoId<EmotePrototype>> Emotes = new();
```